.travis.yml

```diff
@@ -14,7 +14,9 @@ branches:
 install:
   - rm -rf public || exit 0
   - wget -O hugo.tar.gz https://github.com/gohugoio/hugo/releases/download/v0.37.1/hugo_0.37.1_Linux-64bit.tar.gz && tar xzf hugo.tar.gz
+  - wget https://github.com/mikefarah/yq/releases/download/v4.2.0/yq_linux_amd64 -O /usr/bin/yq && chmod +x /usr/bin/yq
 script:
+  - yq eval -j data/mapping.yml >> content/post/mirror-help/mapping.html
```

Comment:
Why does this line write JSON into an html file (mapping.html)?

Reply:
The resulting `mirror-help.xml` will then have a special entry `metadata:mapping`. Inside that is JSON data. We could parse it in frontend.

Reply:
Well, this looks like red herring to me despite possible escaping issues:

```
    <item>
      <title>metadata:mapping</title>
      <link>https://sjtug.org/post/mirror-help/mapping/</link>
      <pubDate>Mon, 01 Jan 0001 00:00:00 +0000</pubDate>
      
      <guid>https://sjtug.org/post/mirror-help/mapping/</guid>
      <description>{
  &#34;groups&#34;: {
    &#34;rust&#34;: {
      &#34;name&#34;: &#34;Rust 语言及工具链&#34;,
      &#34;repos&#34;: [
        &#34;crates.io&#34;,
        &#34;git/crates.io-index&#34;,
        &#34;static.crates.io&#34;,
        &#34;rust-static&#34;
      ]
    },

```

Better to just output JSON into `static/mapping.json` and fetch the file from the frontend.